src/Autocomplete.js

```diff
@@ -149,6 +245,16 @@ class Autocomplete extends AutocompleteBase {
 	}
 
 	/**
+	 * Removes the CSS class from that current selected element.
+	 * @protected
+	 */
+	unSelectCurrentFocusedItem_() {
```

Comment:
Instead of having this be a separate function, it seems like it'd make more sense to add this logic inside `selectItem_`, thus making sure that whenever we select a new item, the previous one is deselected first. I think that would cover all cases, since we always have at least one selected item whenever the list is showing up.

By the way, I've come across a bug related to this when running this code. To reproduce it you basically just need to type `a` in the autocomplete input, then move down with the arrow key to the second option, then type an `l` in the input. That will select the first item of the list again, but it won't deselect the second one first. This gif shows it happening: 

![screen recording 2016-11-01 at 01 29 pm](https://cloud.githubusercontent.com/assets/5216049/19897625/74e01b70-a037-11e6-9e6d-0b6b434f5815.gif)

If this logic is moved inside `selectItem_` this shouldn't happen anymore.